numpy/linalg/_linalg.py

```diff
@@ -815,6 +828,8 @@ def cholesky(a):
     with errstate(call=_raise_linalgerror_nonposdef, invalid='call',
                   over='ignore', divide='ignore', under='ignore'):
         r = gufunc(a, signature=signature)
+    if upper:
+        r = matrix_transpose(r).conj()
```

Comment:
Now wondering about this, since this makes a copy of a potentially large matrix: an alternative would be to actually define `cholesky_up` (like  done for eigenvalues, etc.), i.e., creating a new gufunc by adding to 

https://github.com/numpy/numpy/blob/0032ede015c9b06f88cc7f9b07138ce35f4357ae/numpy/linalg/umath_linalg.cpp#L1988-L1993

https://github.com/numpy/numpy/blob/0032ede015c9b06f88cc7f9b07138ce35f4357ae/numpy/linalg/umath_linalg.cpp#L4178

 https://github.com/numpy/numpy/blob/0032ede015c9b06f88cc7f9b07138ce35f4357ae/numpy/linalg/umath_linalg.cpp#L4390-L4399

Reply:
If it's Ok, let me create later a separate issue/PR that introduces a new ufunc.

Reply:
Sure, it is fine to have the support now and improve it later!